lib/active_merchant/billing/gateways/payu_in.rb

```diff
@@ -0,0 +1,234 @@
+# encoding: utf-8
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class PayuInGateway < Gateway
+      self.test_url = "https://test.payu.in/_payment"
+      self.live_url = "https://secure.payu.in/_payment"
+
+      TEST_INFO_URL = "https://test.payu.in/merchant/postservice.php?form=2"
+      LIVE_INFO_URL = "https://info.payu.in/merchant/postservice.php?form=2"
+
+      self.supported_countries = ['IN']
+      self.default_currency = 'INR'
+      self.supported_cardtypes = [:visa, :master, :american_express, :diners_club]
+
+      self.homepage_url = 'https://www.payu.in/'
+      self.display_name = 'PayU India'
+
+      def initialize(options={})
+        requires!(options, :key, :salt)
+        super
+      end
+
+      def purchase(money, payment, options={})
+        requires!(options, :order_id)
+
+        post = {}
+        add_invoice(post, money, options)
+        add_payment(post, payment)
+        add_addresses(post, options)
+        add_customer_data(post, options)
+        add_auth(post)
+
+        MultiResponse.run do |r|
+          r.process{commit(url("purchase"), post)}
+          if(r.params["enrolled"].to_s == "0")
+            r.process{commit(r.params["post_uri"], r.params["form_post_vars"])}
+          else
+            r.process{handle_3dsecure(r)}
+          end
+        end
+      end
+
+      def refund(money, authorization, options={})
+        raise ArgumentError, "Amount is required" unless money
+
+        post = {}
+
+        post[:command] = "cancel_refund_transaction"
+        post[:var1] = authorization
+        post[:var2] = generate_unique_id
+        post[:var3] = amount(money)
+
+        add_auth(post, :command, :var1)
+
+        commit(url("refund"), post)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(/(ccnum=)[^&\n"]*(&|\n|"|$)/, '\1[FILTERED]\2').
+          gsub(/(ccvv=)[^&\n"]*(&|\n|"|$)/, '\1[FILTERED]\2').
+          gsub(/(card_hash=)[^&\n"]*(&|\n|"|$)/, '\1[FILTERED]\2').
```

Comment:
@rwdaigle I'm considering it sensitive since I don't know how they've constructed it; if it's done naively, then it's probably trivially reversible.